### DISCUSSION TITLE
"Modes" in Behavior Trees

### DISCUSSION THREAD

**Discussion by varungiridhar** [BehaviorTree/BehaviorTree.CPP#616] [Q&A]
Hi,

I would like to create a behavior tree that allows me to essentially execute a certain mode based on user input. For example:

This was my initial design choice:
```
fallback
├── sequence
│   ├── check if mode == 1
│   └── execute mode #1 subtree
└── sequence
    ├── check if mode == 2
    └── execute mode #2 subtree
// assuming the output of the tree does not matter.
```

However, this basically encodes two if-then operators - a design paradigm that does not align well with the PPA logic recommended with behavior trees.

I also realize that there is a difference between such "modes" and post-conditions i.e. modes are not post-conditions; rather, they are conditions required for the execution of a subtree.

I would appreciate any feedback or comments on whether the above bt structure is viable and whether there are any other design choices that carry the same advantages of the reactivity of BT's.

Thanks,
Varun

**Comment by facontidavide** [comment]
You have many ways to implement this:

- **IfTheElse** node
- **Switch** node
- Plain Sequence with **Preconditions**
